src/plugins/query_enhancements/public/query_assist/components/query_assist_summary.tsx

```diff
@@ -206,6 +207,19 @@ export const QueryAssistSummary: React.FC<QueryAssistSummaryProps> = (props) =>
     });
   }, [props.core]);
 
+  useEffect(() => {
+    const checkSummaryAgent = async () => {
+      setIsSummaryAgent(false);
+      if (selectedDataset.current?.dataSource?.id) {
+        const res = await getIsSummaryAgent(selectedDataset.current.dataSource.id);
```

Comment:
Should we check if `assistantClient` was assigned before calling `getIsSummaryAgent`?